src/IO/ADIOS/ParallelADIOS1IOHandler.cpp

```diff
@@ -58,19 +58,26 @@ ParallelADIOS1IOHandlerImpl::~ParallelADIOS1IOHandlerImpl()
 
     if( this->m_handler->accessTypeBackend != AccessType::READ_ONLY )
     {
+
         for( auto& group : m_attributeWrites )
             for( auto& att : group.second )
                 flush_attribute(group.first, att.first, att.second);
 
-        /* create all files, even if ADIOS file creation has been deferred,
-         * but execution of the deferred operation has never been triggered
-         * (happens when no Operation::WRITE_DATASET is performed) */
-        for( auto& f : m_filePaths )
-            if( m_openWriteFileHandles.find(f.second) == m_openWriteFileHandles.end() )
-                m_openWriteFileHandles[f.second] = open_write(f.first);
-
+    // unordered map caused the value of the same container
+    // stored with different orders in different processors.
+    // which  caused trouble with  close(), which is collective
+    // so I just sort by file name to force  all processors  close
+    // all the fids in the same order
+    std::map< std::string, int64_t> allFiles;
```

Comment:
The next three lines can probably be [shortened to](https://en.cppreference.com/w/cpp/container/map/map):
```suggestion
    std::map< std::string, int64_t> allFiles(
        m_openWriteFileHandles.begin(), m_openWriteFileHandles.end() );
```

Reply:
I would prefer to stay put.